todo_app/templates/index.html

```diff
@@ -2,16 +2,82 @@
       <h2>Items</h2>
-      <ul class="list-group mb-4">
-        <!-- List the todo items here -->
+      <ul class="list-group shadow">
+        {% for item in items %}
+          {% set theme = item_status_data[item.status].theme %}
+          <li class="list-group-item bg-light text-dark d-flex justify-content-between align-items-center px-3">
+            <div class="text-break">{{ item.title }}</div>
+            <div class="d-inline-flex flex-sm-row flex-column-reverse align-items-sm-center align-items-end">
+              <span class="btn-group text-nowrap ml-3 my-1">
+                <strong class="btn btn-sm btn-outline-{{ theme }} disabled" style="width: 6rem; opacity: 1;">
+                  {{ item.status }}
+                </strong>
+                <button type="button" class="btn btn-sm btn-{{ theme }} dropdown-toggle dropdown-toggle-split"
+                        data-toggle="dropdown" data-display="static" aria-expanded="false">
+                  <span class="sr-only">Toggle Dropdown</span>
+                </button>
+                <div class="dropdown-menu border-{{ theme }} w-100 py-1" style="min-width: unset;">
+                  {% for item_status in item_status_data.keys() %}
+                    {% set is_current_status = item_status == item.status %}
+                    <a class="dropdown-item small text-right pl-3 {{ 'disabled' if is_current_status }}"
+                       href="{{ url_for('update_item_status', item_id=item.id, item_status=item_status) }}"
+                       {% if is_current_status %} aria-disabled="true" tabindex="-1" {% endif %}>
+                      {{ item_status }}
+                    </a>
+                  {% endfor %}
+                  <span role="presentation" class="position-absolute w-100 h-25"></span>
+                </div>
+              </span>
+              <button type="button" class="close text-{{ theme }} ml-3 mb-1" aria-label="Close"
+                      data-toggle="modal" data-target="#modal-{{ item.id }}">
+                <span class="align-baseline" aria-hidden="true">&times;</span>
+              </button>
+            </div>
+            <div class="modal fade" id="modal-{{ item.id }}" tabindex="-1" aria-hidden="true">
+              <div class="modal-dialog">
+                <div class="modal-content">
+                  <div class="modal-header">
+                    <h5 class="modal-title">Confirm deletion</h5>
+                    <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                      <span aria-hidden="true">&times;</span>
+                    </button>
+                  </div>
+                  <div class="modal-body">
+                    <p class="list-group-item bg-light rounded">{{ item.title }}</p>
+                    <p>Are you sure you want to delete this task?</p>
+                  </div>
+                  <div class="modal-footer">
+                    <a role="button" class="btn btn-danger"
```

Comment:
These look great, but I'd still be inclined to use forms/POST requests for the deleting/updating actions rather than GET requests in order to stick to a more semantic approach

Reply:
Thanks! Hadn't thought about this before, but I agree that it's much better this way